### DISCUSSION TITLE
Stealth -> Xip... Quo Vadis?

### DISCUSSION THREAD

**Discussion by svoop** [hellostealth/stealth#296] [Q&A]
In order to migrate my chatops away from Keybase, I'm playing around with Stealth (2-beta). There's no service for Matrix yet, I'd be happy to give it a shot.

However, I'm not sure where Stealth is going. Apparently, there's a migration from Stealth to Xip underway, however, the repo for the latter is unavailable and there is minimal activity on the Stealth repo.

@mgomes Since you're the owner, can you give an update or roadmap as to where this is all going and what people like me should do:

* [ ] use Stealth and at some point migrate to Xip (app and services)
* [ ] look elsewhere
* [ ] wait (well, not really an option)

Thanks for the clarification!

**Comment by mgomes** [comment]
@svoop thanks for reaching out! Stealth (and particularly the 2.0 beta) has some nice features for making advanced bots that will stick around regardless of other changes. One of the design shortcomings I am trying to address with Xip is handling messaging services that aren't strictly chat based (Slack for example). So handling other events such as likes, reactions, etc. Right now that is kind of difficult to conceptually fit into the current architecture.

@matthewblack might be able to provide more context with respect to these things and Stealth, though.

Can I ask, are you arriving at Stealth because you have existing Ruby experience?

**Reply by svoop** [reply]
Thanks for the update, @mgomes. As for me, yes, when it comes to my open source voluntary work, I'm all Ruby.

I came across Stealth looking for the foundation to implement chatops for openflightmaps.org. Unfortunately, another problem has come up: Our infra guy prefers the chatops to be on Google Cloud Run or some Kubernetes-based pay per demand solution. That's a lot of overhead for a rather simple tool and I'm quite sure, it's easier to build a stateless chatbot on Sinatra than rolling out mostly unnecessary cloud power to build it on Stealth.
